test/rules/no-overlapping-elements.js

```diff
@@ -0,0 +1,63 @@
+import RuleTester from '../../lib/testers/rule-tester';
+
+import rule from '../../rules/no-overlapping-elements';
+
+import {
+  readModdle
+} from '../../lib/testers/helper';
+
+
+RuleTester.verify('no-overlapping-elements', rule, {
+  valid: [
+    {
+      moddleElement: readModdle(__dirname + '/no-overlapping-elements/valid-boundary-event.bpmn')
+    },
+    {
+      moddleElement: readModdle(__dirname + '/no-overlapping-elements/valid-collaboration.bpmn')
+    },
+    {
+      moddleElement: readModdle(__dirname + '/no-overlapping-elements/valid-single-process.bpmn')
+    }
+  ],
+  invalid: [
+    {
+      moddleElement: readModdle(__dirname + '/no-overlapping-elements/invalid-boundary-event.bpmn'),
+      report: [
+        {
+          id: 'Event_1qrxqaa',
+          message: 'Element overlaps with other element'
+        },
+        {
+          id: 'Activity_1r8gugk',
+          message: 'Element overlaps with other element'
+        }
+      ]
+    },
+    {
+      moddleElement: readModdle(__dirname + '/no-overlapping-elements/invalid-collaboration.bpmn'),
+      report: [
+        {
+          id: 'Participant_1af2hnr',
+          message: 'Element overlaps with other element'
```

Comment:
Do we want to change this to an ID reference, i.e. report `Element overlaps with <#someID>` if the other element has an ID? It could be important context to fix the issue.

Reply:
Yes I can change it to ```Element overlaps with <#someID>```, but how should it behave when more than one overlap is present? Should it be ```Element overlaps with <List_of_IDs>``` or separated report message for each overlap?

Reply:
I see how that may be a little bit more complicated. Let's take it as a follow up. :)

Reply:
Ok, so how should that ```AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:``` error be fixed?
I can't see any errors in tests.

Reply:
I'll have a look into this and get back to you.

Reply:
@david-d-le The error is properly colored for me in the console.

Essentially your rule does not seem to work as expected yet, not resulting in the desired lint output: 

![image](https://github.com/bpmn-io/bpmnlint/assets/58601/8d60acfe-1d77-41ee-a547-20d73ce14561)

Reply:
ok i will look at it :)

Reply:
Still no idea where the issue is. The rule is working in the [bpmnlint-playground](https://github.com/bpmn-io/bpmnlint-playground)

Reply:
The `bpmnlint-playground` actually used ancient bpmn-js dependencies.

With the latest version (https://github.com/bpmn-io/bpmnlint-playground/blob/master/CHANGELOG.md#050) you'll see the same behavior across Node and browser.

Do not rely on `businessObject.di` but instead build your own `di` map and access the graphical information from there.

Reply:
Oh I didn't know that. So if I understand correctly, I need to find di information from ```Definitions.Diagrams``` manually?

Reply:
Exactly.

Reply:
solved